unable to install

Operating System Version:
Fedora 28

Popcorn Time Version:
latest as of 11 July 2019

Download date:
11 July 2019

#### Expected Behaviour
complete installation upon execution of  "./make_popcorn.sh" in terminal
...

#### Actual Behaviour
Returns with "permission denied" in terminal
...

#### Steps to reproduce the behaviour

1. open terminal
2. log in as root
3. enter directory where "make_popcorn.sh" is located and attempt to execute the make file

I made a little progress. So I didn't allow for `make_popcorn.sh` to be run as an executable, so I checked the box in the properties. I'm missing a `gulp-load-plugins module`.

`[akollipara@localhost popcorn-desktop-c7799212b5a4779a7b34ad2d9a28632e50e3ae55]$ ./make_popcorn.sh
Cloning Popcorn Time
Where do you wish to clone Popcorn Time to? [popcorn] 
Cloning into 'popcorn'...
remote: Enumerating objects: 41833, done.
remote: Total 41833 (delta 0), reused 0 (delta 0), pack-reused 41833
Receiving objects: 100% (41833/41833), 32.01 MiB | 3.45 MiB/s, done.
Resolving deltas: 100% (28142/28142), done.
Cloned Popcorn Time successfully
Switched to /home/akollipara/Downloads/popcorn-desktop-c7799212b5a4779a7b34ad2d9a28632e50e3ae55/popcorn
module.js:550
    throw err;
    ^

Error: Cannot find module 'gulp-load-plugins'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/akollipara/Downloads/popcorn-desktop-c7799212b5a4779a7b34ad2d9a28632e50e3ae55/popcorn/gulpfile.js:16:11)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
Popcorn Time encountered an error and couldn't be built
`

I'm also noticing there are several modules, like `run-sequence`, `del`, or `gulp-download2` that can't be found. I'll install them with npm and see where I end up.

--- a/make_popcorn.sh
+++ b/make_popcorn.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 ## Version 0.1.1
 ##
@@ -22,17 +22,6 @@ elif [ "$1" = "ssh" ]; then
 else
     clone_url="$1"
 fi
-
-execsudo() {
-    case $OSTYPE in msys*)
-       echo $OSTYPE
-       $1
-       ;;
-    *)
-       sudo $1
-       ;;
-    esac
-}
 
 clone_command() {
     if git clone $clone_url $dir; then
@@ -109,10 +98,7 @@ if [ "$clone_repo" = "True" ]; then
             echo "Removing old directory"
             if [ "$dir" != "." ] || [ "$dir" != "$PWD" ]; then
                 echo "Cleaning up from inside the destination directory"
-                sudo rm -rf $dir/*
-            else
-                echo "Cleaning up from outside the destination directory"
-                sudo rm -rf $dir
+                rm -rf $dir
             fi
             clone_command
         else
@@ -120,23 +106,6 @@ if [ "$clone_repo" = "True" ]; then
         fi
     fi
 fi
-try="True"
-tries=0
-while [ "$try" = "True" ]; do
-    read -p "Do you wish to install the required dependencies for Popcorn Time and setup for building? (yes/no) [yes] " rd_dep
-    if [ -z "$rd_dep" ]; then
-        rd_dep="yes"
-    fi
-    tries=$((tries+1))
-    if [ "$rd_dep" = "yes" ] || [ "$rd_dep" = "no" ]; then
-        try="False"
-    elif [ "$tries" -ge "3" ]; then
-        echo "No valid input, exiting"
-        exit 3
-    else
-        echo "Not a valid answer, please try again"
-    fi
-done
 
 if [ -z "$dir" ]; then
     dir="."
@@ -146,60 +115,19 @@ echo "Switched to $PWD"
 
 if [ "$rd_dep" = "yes" ]; then
 
-    echo "Installing global dependencies"
-    if execsudo "npm install -g bower gulp gulp-cli"; then
-        echo "Global dependencies installed successfully!"
-    else
-        echo "Global dependencies encountered an error while installing"
-        exit 4
-    fi
-
-    echo "Downloading Popcorn dependencies files"
-    if execsudo "wget -i popcorn_dependencies_files.txt"; then
-        echo "Local Popcorn files downloaded successfully!"
-        echo "Moving Popcorn local files"
-        mv torrent_collection.js src/app/lib/views/torrent_collection.js
-    else
-        echo "Local Popcorn files encountered an error while downloading"
-        exit 4
-    fi
-
     echo "Installing local dependencies"
-    if execsudo "npm install"; then
-        echo "Local dependencies installed successfully!"
-    else
-        echo "Local dependencies encountered an error while installing"
-        exit 4
-    fi
-
-    curh=$HOME
-    case $OSTYPE in msys*)
-        ;;
-        *)
-        if execsudo "chown -R $USER ." && execsudo "chown -R $USER $curh/.cache"; then
-            echo "Local permissions corrected successfully!"
-        else
-            echo "Local permissions encountered an error while correcting"
-            exit 4
-        fi
-        ;;
-    esac
-
-    echo "Setting up Bower"
-    if bower install; then
-        echo "Bower successfully installed"
-    else
-        echo "Encountered an error while installing bower"
-        exit 4
-    fi
-
+    yarn config set yarn-offline-mirror ./node_modules/
+    yarn install --ignore-engines
+    yarn build
     echo "Successfully setup for Popcorn Time"
 fi
 
 if gulp build; then
     echo "Popcorn Time built successfully!"
-    ./Create-Desktop-Entry
-    echo "Run 'gulp run' from inside the repository to launch the app"
+    if [[ `uname -s` != *"NT"* ]]; then # if not windows
+        ./Create-Desktop-Entry
+    fi
+    echo "Run 'gulp run' from inside the repository to launch the app or check out under ./build folder..."
     echo "Enjoy!"
 else
     echo "Popcorn Time encountered an error and couldn't be built"

